### ISSUE TITLE
User must sign in again after on hour

### ISSUE THREAD

**Issue by pekspro** [microsoftgraph/aspnetcore-connect-sample#20]
When running this sample, it looks like the user must sign in once per hour (maybe after hour of inactivity). I know that the Azure authentication token expires after one hour, so I guess that is related.

Is there some easy way to fix this? If I create a new application with Azure sign in the user is signed in a lot longer by default.

I have upgraded this sample to use Asp.Net Core 2.1, and I think it behaves the same as in version 2.0.

(Otherwise I think the sample is great :-)

**Comment by mark-szabo** [comment]
Hi @pekspro,
So basically you cannot alter AzureAD's authentication settings, so you have no choice but to redirect the user to the login page after the token has expired.
With ADAL (Azure Authentication Library - v1) after this redirect the token was given without promting the user if the user was still signed in. With MSAL (Microsoft Authentication Library - v2) the user needs to choose the account he/se want's to log in width but does not needs to enter password/sign in again. 
In theory you can get around it with using `login_hint`: https://github.com/microsoftgraph/aspnetcore-connect-sample/issues/8#issuecomment-347266261

**Comment by KalyanChanumolu-MSFT** [comment]
The Access token expires every 60 minutes. This is the default behavior.
It was supposed to be made [configurable ](https://docs.microsoft.com/en-us/azure/active-directory/active-directory-configurable-token-lifetimes) but that feature is going away in favor of [Azure Active Directory Conditional Access](https://docs.microsoft.com/en-us/azure/active-directory/conditional-access/overview)

**Comment by pekspro** [comment]
Thanks for all answers :-)

I tried to use login hint as suggested in #8. But I was still forced to select my account.

So instead I created an iframe on a page where the embedded page (with the [Authorize] attribute) was reloaded every minute. This seemed to keep the authentication alive. That’s a pretty good solution.

I also tried to do the solution suggested on: http://www.cloudidentity.com/blog/2016/07/25/controlling-a-web-apps-session-duration-2/

This also uses an embedded page in an iframe but forces a new token on each request. The solution above doesn’t use ASP.NET Core, so I ported the ForcedSignIn page with the following:

        public async Task<IActionResult> ForcedSignIn()
        {
            // Send an OpenID Connect sign -in request.
            await HttpContext.ChallengeAsync(new AuthenticationProperties()
            {
                RedirectUri = "/Home/Contact"
            }
            );

            return View();
        }

Not sure if that’s correct, but I hope so :-) Either way, that didn’t work either. When doing this I was redirected to an authentication page that wasn’t allowed to be shown in an iframe. If I browsed to ForcedSignIn page without using an iframe I was still forced to select my account. 

I’m a bit confused by my last attempt since I would expect the same thing happen if I just had a page reloaded every minute.
